packages/provider/src/RelayedTransactionValidator.ts

```diff
@@ -23,77 +63,130 @@ export class RelayedTransactionValidator {
+    let isTransactionTypeValid = true
+    let isFeeMarket1559Transaction = false
+    let isLegacyGasPriceValid = false
+    let isMaxFeePerGasValid = false
+    let isMaxPriorityFeePerGasValid = false
 
-    if (
-      isSameAddress(transaction.to, relayHubAddress) &&
-      relayRequestAbiEncode === transaction.data &&
-      isSameAddress(request.relayRequest.relayData.relayWorker, signer)
-    ) {
-      if (transaction.gasPrice != null) {
-        if (toBN(transaction.gasPrice).lt(toBN(request.relayRequest.relayData.maxPriorityFeePerGas))) {
-          throw new Error(`Relay Server signed gas price too low (${transaction.gasPrice}). Requested transaction with gas price at least ${request.relayRequest.relayData.maxPriorityFeePerGas}`)
-        }
-      } else if (transaction.maxFeePerGas != null && transaction.maxPriorityFeePerGas != null) {
-        if (toBN(transaction.maxPriorityFeePerGas).lt(toBN(request.relayRequest.relayData.maxPriorityFeePerGas))) {
-          throw new Error(`Relay Server signed max priority fee too low (${transaction.maxPriorityFeePerGas}). Requested transaction with priority fee at least ${request.relayRequest.relayData.maxPriorityFeePerGas}`)
-        }
-        if (toBN(transaction.maxFeePerGas).lt(toBN(request.relayRequest.relayData.maxFeePerGas))) {
-          throw new Error(`Relay Server signed max fee too low (${transaction.maxFeePerGas}). Requested transaction with max fee at least ${request.relayRequest.relayData.maxFeePerGas}`)
-        }
-      } else {
-        throw new Error('Transaction must have either gasPrice or (maxFeePerGas and maxPriorityFeePerGas)')
-      }
-      // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
-      const receivedNonce = parseInt(transaction.nonce!)
-      if (receivedNonce > request.metadata.relayMaxNonce) {
-        // TODO: need to validate that client retries the same request and doesn't double-spend.
-        // Note that this transaction is totally valid from the EVM's point of view
-
-        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-        throw new Error(`Relay used a tx nonce higher than requested. Requested ${request.metadata.relayMaxNonce} got ${receivedNonce}`)
-      }
-
-      this.logger.info('validateRelayResponse - valid transaction response')
-      return true
+    if (transaction instanceof Transaction) {
+      isLegacyGasPriceValid = transaction.gasPrice.gte(toBN(request.relayRequest.relayData.maxFeePerGas))
+      // throw new Error(`Relay Server signed gas price too low (${transaction.gasPrice}). Requested transaction with gas price at least ${request.relayRequest.relayData.maxPriorityFeePerGas}`)
+    } else if (transaction instanceof FeeMarketEIP1559Transaction) {
+      isFeeMarket1559Transaction = true
+      isMaxPriorityFeePerGasValid = transaction.maxPriorityFeePerGas.gte(toBN(request.relayRequest.relayData.maxPriorityFeePerGas))
```

Comment:
sshould compare to latest ping result's maxPriorityFee.
(if relayer returned 5, and I decided on prioFee of 10, I can't expect anyone else to have that prio, only what the server's minimum is)